sqlglot/dialects/snowflake.py

```diff
@@ -327,3 +327,10 @@ def generatedasidentitycolumnconstraint_sql(
             increment = expression.args.get("increment")
             increment = f" INCREMENT {increment}" if increment else ""
             return f"AUTOINCREMENT{start}{increment}"
+
+        def create_sql(self, expression: exp.Create) -> str:
+            unsupported = ["set", "multiset", "global_temporary", "volatile"]
+            for prop in unsupported:
+                expression.set(prop, False)
```

Comment:
We shouldn't mutate any expression in the generator. Whenever something like this is needed, you first need to copy it.